src/libraries/System.Security.Cryptography.Xml/ref/System.Security.Cryptography.Xml.cs

```diff
@@ -80,6 +80,7 @@ public abstract partial class EncryptedReference
         protected EncryptedReference() { }
         protected EncryptedReference(string? uri) { }
         protected EncryptedReference(string? uri, System.Security.Cryptography.Xml.TransformChain transformChain) { }
+        [System.Diagnostics.CodeAnalysis.MemberNotNullWhenAttribute(true, (string)null!)]
```

Comment:
Why is second argument `null`? You can hardcode private member

Reply:
I got an error when I used an actual name (or any non null value). Plus, I thought that null (as in, completely empty implementations) is the way to do it in refs?

Reply:
`MemberNotNull` attributes shouldn't be present in the ref.cs unless they're describing a public or protected field.

Reply:
`_cachedXml` is `internal`. I removed the attribute from this file, but got the following compilation error:

` error : CannotRemoveAttribute : Attribute 'System.Diagnostics.CodeAnalysis.MemberNotNullWhenAttribute' exists on 'System.Security.Cryptography.Xml.EncryptedReference.CacheValid.get()' in the implementation but not the reference`.

So I took a educated guess from that error message and put the attribute on the getter instead of the overall property, and there are no compilation errors.

Reply:
Hm, why does this `ref` contains internal members at all? (`CacheValid`)

Reply:
Even on the getter, it's not an attribute we want in the ref.cs file, since it's not describing anything that can be observed by a caller.  If everything is building except this one item failing apicompat, you should be able to build like `dotnet msbuild /p:UpdateBaselineFiles=true` and it should add an exception for this item in the baseline file (which it might have to create).

> Hm, why does this ref contains internal members at all? (CacheValid)

`CacheValid` is `protected internal` which is `protected OR internal`.  I'm not 100% positive why we don't transform that to `protected` in the ref.cs files, but... we don't :)

Reply:
I remove the attribute and ran `msbuild /p:UpdateBaselineFiles=true`. The errors from earlier appeared again. So I did a `dotnet clean`, and ran it again, and no errors! \o/  Thanks for your help.

Reply:
@bartonjs - sorry, I'm really stumped on this one. I'm getting intermittent build errors. I can run `msbuild /p:UpdateBaselineFiles=true` one time, and get build errors, and run it again, and no build errors, and then run `dotnet build` and get build errors. I did try to research what's going on, but I can't find any documentation on `UpdateBaselineFiles`.  Any advice?  Currently, my branch in the local repo is identical to the branch on remote repo. My local branch builds fine, but I can see build errors in the PR.

Reply:
Just to be clear that I'm doing the right thing: `CacheValid` **should not** be in the ref files? Or it **should** be in the ref files, but just listed as `protected` rather than `protected internal`? When I remove it from the ref file, everything builds fine.

Reply:
Added back. I still get the build errors, but I seem to get them regardless of what's in the file.